hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/BaseTokenHandler.java

```diff
@@ -114,6 +114,9 @@ protected void changeSupply(
         requireNonNull(token);
         requireNonNull(treasuryRel);
         requireNonNull(invalidSupplyCode);
+        if (token.hasKycKey()) {
+            validateTrue(treasuryRel.kycGranted(), ACCOUNT_KYC_NOT_GRANTED_FOR_TOKEN);
+        }
```

Comment:
I don't see this added logic anywhere in [the old code](https://github.com/hashgraph/hedera-services/blob/develop/hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/store/models/Token.java#L359). I'm not sure this is correct